### PR TITLE
Solve the problem that Span cannot be closed normally when Http.Client is a singleton

### PR DESCRIPTION
**Pull request by Matrix-K**
As the title says

### CODE REVIEW

File: plugins/http/client.go [hunk 1]
```diff
@@ -116,5 +116,6 @@ func (t *transport) RoundTrip(req *http.Request) (res *http.Response, err error)
 	if res.StatusCode >= http.StatusBadRequest {
 		span.Error(time.Now(), "Errors on handling client")
 	}
+	span.End()
```
Comment (arugal) [inline plugins/http/client.go:119]: image

Hi @Matrix-K I don't see what you mean. In my opinion, this creates new bugs

**Review by arugal — CHANGES_REQUESTED** [review]

File: plugins/http/client.go (hunk 1 above)
Comment (Matrix-K) [reply plugins/http/client.go:119]: Sorry, I forgot to close the span when roundTrip error.   I'll solve this later.

What I mean is that when the http.Client connection pool is used globally, defer span.End() cannot be closed normally

Just like Image below

image

Comment (arugal) [reply plugins/http/client.go:121]: @Matrix-K Can you provide me with a sample project? I think your modified code has the same effect as using `defer` :)

**Comment by arugal** [comment]
@Matrix-K any update?

**Comment by wu-sheng** [comment]
@Matrix-K We need your update, otherwise, we have to close this PR.

**Pull request closed without merging**